server/service/devices.go

```diff
@@ -0,0 +1,157 @@
+package service
+
+import (
+	"context"
+
+	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
+	hostctx "github.com/fleetdm/fleet/v4/server/contexts/host"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+)
+
+/////////////////////////////////////////////////////////////////////////////////
+// Get Current Device's Host
+/////////////////////////////////////////////////////////////////////////////////
+
+type getDeviceHostRequest struct {
+	Token string `url:"token"`
+}
+
+func (r *getDeviceHostRequest) deviceAuthToken() string {
+	return r.Token
+}
+
+func getDeviceHostEndpoint(ctx context.Context, request interface{}, svc fleet.Service) (interface{}, error) {
+	host, ok := hostctx.FromContext(ctx)
+	if !ok {
+		err := ctxerr.Wrap(ctx, fleet.NewAuthRequiredError("internal error: missing host from request context"))
+		return getHostResponse{Err: err}, nil
+	}
+
+	// must still load the full host details, as it returns more information
+	hostDetails, err := svc.GetHost(ctx, host.ID)
+	if err != nil {
+		return getHostResponse{Err: err}, nil
+	}
+
+	resp, err := hostDetailResponseForHost(ctx, svc, hostDetails)
+	if err != nil {
+		return getHostResponse{Err: err}, nil
+	}
+
+	return getHostResponse{Host: resp}, nil
+}
+
+// AuthenticateDevice returns the host identified by the device authentication
+// token, along with a boolean indicating if debug logging is enabled for that
+// host.
+func (svc *Service) AuthenticateDevice(ctx context.Context, authToken string) (*fleet.Host, bool, error) {
+	// skipauth: Authorization is currently for user endpoints only.
+	svc.authz.SkipAuthorization(ctx)
+
+	if authToken == "" {
+		return nil, false, ctxerr.Wrap(ctx, fleet.NewAuthRequiredError("authentication error: missing device authentication token"))
+	}
+
+	host, err := svc.ds.LoadHostByDeviceAuthToken(ctx, authToken)
+	switch {
+	case err == nil:
+		// OK
+	case fleet.IsNotFound(err):
+		return nil, false, ctxerr.Wrap(ctx, fleet.NewAuthRequiredError("authentication error: invalid device authentication token"))
+	default:
+		return nil, false, ctxerr.Wrap(ctx, err, "authenticate device")
+	}
+
+	// TODO: do we want that seen time update for device auth token? It's not "seen"
+	// in the sense that it's not osquery that did ping back to fleet, so I think we
+	// might want to remove that from here?
+
+	//// Update the "seen" time used to calculate online status. These updates are
+	//// batched for MySQL performance reasons. Because this is done
+	//// asynchronously, it is possible for the server to shut down before
+	//// updating the seen time for these hosts. This seems to be an acceptable
+	//// tradeoff as an online host will continue to check in and quickly be
+	//// marked online again.
+	//svc.seenHostSet.addHostID(host.ID)
+	//host.SeenTime = svc.clock.Now()
```

Comment:
I think we want to remove this "seen host" update from this endpoint, but just to be sure ^

Reply:
+1 to this, seen time should be only handled by the "osquery side".